<!DOCTYPE html>
<html lang="en-us">
  <head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=443&amp;path=livereload" data-no-instant defer></script>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>Gene ID Conversion | Antonio Ahn&#39;s website</title>
    <link rel="stylesheet" href="../../css/style.css" />
    <link rel="stylesheet" href="../../css/fonts.css" />
    
  </head>

  <body>
    <nav>
    <ul class="menu">
      
      <li><a href="../../">Home</a></li>
      
      <li><a href="../../posts/">Posts</a></li>
      
      <li><a href="../../about/">About</a></li>
      
      <li><a href="../../categories/">Categories</a></li>
      
      <li><a href="../../tags/">Tags</a></li>
      
      <li><a href="../../publication/">Publications</a></li>
      
      <li><a href="../../resume/">Resume</a></li>
      
    </ul>
    <hr/>
    </nav>

<div class="article-meta">
<h1><span class="title">Gene ID Conversion</span></h1>

<h2 class="date">2024/01/07</h2>
</div>

<main>
<p>Various ways to convert ENTREZ ID to gene symbols and vice versa</p>
<pre><code class="language-r">library(tidyverse)
library(DOSE)
</code></pre>
<h1 id="load-dataset">load dataset</h1>
<pre><code class="language-r"># load a list of ENTREZ ID's
data(geneList)
geneList %&gt;% head
</code></pre>
<pre><code>##     4312     8318    10874    55143    55388      991 
## 4.572613 4.514594 4.418218 4.144075 3.876258 3.677857
</code></pre>
<pre><code class="language-r">length(names(geneList))
</code></pre>
<pre><code>## [1] 12495
</code></pre>
<h1 id="biomart">biomart</h1>
<pre><code class="language-r">library(biomaRt)
</code></pre>
<pre><code class="language-r">biomaRt::listEnsemblArchives()
</code></pre>
<pre><code>##              name     date                                 url version
## 1  Ensembl GRCh37 Feb 2014          https://grch37.ensembl.org  GRCh37
## 2     Ensembl 110 Jul 2023 https://jul2023.archive.ensembl.org     110
## 3     Ensembl 109 Feb 2023 https://feb2023.archive.ensembl.org     109
## 4     Ensembl 108 Oct 2022 https://oct2022.archive.ensembl.org     108
## 5     Ensembl 107 Jul 2022 https://jul2022.archive.ensembl.org     107
## 6     Ensembl 106 Apr 2022 https://apr2022.archive.ensembl.org     106
## 7     Ensembl 105 Dec 2021 https://dec2021.archive.ensembl.org     105
## 8     Ensembl 104 May 2021 https://may2021.archive.ensembl.org     104
## 9     Ensembl 103 Feb 2021 https://feb2021.archive.ensembl.org     103
## 10    Ensembl 102 Nov 2020 https://nov2020.archive.ensembl.org     102
## 11    Ensembl 101 Aug 2020 https://aug2020.archive.ensembl.org     101
## 12    Ensembl 100 Apr 2020 https://apr2020.archive.ensembl.org     100
## 13     Ensembl 99 Jan 2020 https://jan2020.archive.ensembl.org      99
## 14     Ensembl 98 Sep 2019 https://sep2019.archive.ensembl.org      98
## 15     Ensembl 97 Jul 2019 https://jul2019.archive.ensembl.org      97
## 16     Ensembl 96 Apr 2019 https://apr2019.archive.ensembl.org      96
## 17     Ensembl 95 Jan 2019 https://jan2019.archive.ensembl.org      95
## 18     Ensembl 94 Oct 2018 https://oct2018.archive.ensembl.org      94
## 19     Ensembl 93 Jul 2018 https://jul2018.archive.ensembl.org      93
## 20     Ensembl 80 May 2015 https://may2015.archive.ensembl.org      80
## 21     Ensembl 77 Oct 2014 https://oct2014.archive.ensembl.org      77
## 22     Ensembl 75 Feb 2014 https://feb2014.archive.ensembl.org      75
## 23     Ensembl 54 May 2009 https://may2009.archive.ensembl.org      54
##    current_release
## 1                 
## 2                *
## 3                 
## 4                 
## 5                 
## 6                 
## 7                 
## 8                 
## 9                 
## 10                
## 11                
## 12                
## 13                
## 14                
## 15                
## 16                
## 17                
## 18                
## 19                
## 20                
## 21                
## 22                
## 23
</code></pre>
<pre><code class="language-r">listMarts()
</code></pre>
<pre><code>##                biomart                version
## 1 ENSEMBL_MART_ENSEMBL      Ensembl Genes 110
## 2   ENSEMBL_MART_MOUSE      Mouse strains 110
## 3     ENSEMBL_MART_SNP  Ensembl Variation 110
## 4 ENSEMBL_MART_FUNCGEN Ensembl Regulation 110
</code></pre>
<pre><code class="language-r">mart_aug2020 &lt;- biomaRt::useMart(biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;, 
                         dataset = &quot;hsapiens_gene_ensembl&quot;,
                         host = &quot;https://aug2020.archive.ensembl.org&quot;)

head(listAttributes(mart_aug2020), n = 10)
</code></pre>
<pre><code>##                             name                  description         page
## 1                ensembl_gene_id               Gene stable ID feature_page
## 2        ensembl_gene_id_version       Gene stable ID version feature_page
## 3          ensembl_transcript_id         Transcript stable ID feature_page
## 4  ensembl_transcript_id_version Transcript stable ID version feature_page
## 5             ensembl_peptide_id            Protein stable ID feature_page
## 6     ensembl_peptide_id_version    Protein stable ID version feature_page
## 7                ensembl_exon_id               Exon stable ID feature_page
## 8                    description             Gene description feature_page
## 9                chromosome_name     Chromosome/scaffold name feature_page
## 10                start_position              Gene start (bp) feature_page
</code></pre>
<pre><code class="language-r"># find the entrez gene id name
grep( &quot;entrez&quot;, listAttributes(mart_aug2020)$name, value = T)
</code></pre>
<pre><code>## [1] &quot;entrezgene_trans_name&quot;  &quot;entrezgene_description&quot; &quot;entrezgene_accession&quot;  
## [4] &quot;entrezgene_id&quot;
</code></pre>
<pre><code class="language-r">t2g_aug2020 &lt;- biomaRt::getBM(attributes = c(&quot;entrezgene_id&quot;,&quot;external_gene_name&quot;, &quot;ensembl_gene_id&quot;, &quot;ensembl_gene_id_version&quot;, &quot;gene_biotype&quot;, &quot;transcript_biotype&quot;,&quot;chromosome_name&quot;, &quot;band&quot;, &quot;transcript_length&quot;, &quot;start_position&quot;, &quot;end_position&quot;,&quot;transcription_start_site&quot;, &quot;strand&quot;, &quot;refseq_mrna&quot;,&quot;refseq_ncrna&quot;), mart = mart_aug2020)

# rename
t2g_aug2020 &lt;- dplyr::rename(t2g_aug2020,  entrez_gene = entrezgene_id, ext_gene = external_gene_name, ens_gene = ensembl_gene_id, ens_gene_ver = ensembl_gene_id_version)
</code></pre>
<pre><code class="language-r"># biomart genesymbols
index &lt;- match(names(geneList), t2g_aug2020$entrez_gene)
biomart_gs &lt;- t2g_aug2020[index,] %&gt;% .$ext_gene
# wasn't able to use this because the NA's from match is ignored
# biomart_gs &lt;- t2g_aug2020 %&gt;% slice(match(names(geneList), entrez_gene)) %&gt;% .$ext_gene
</code></pre>
<h1 id="annotationhub">annotationhub</h1>
<pre><code class="language-r">library(AnnotationHub)
</code></pre>
<pre><code class="language-r"># these codes are from https://github.com/hbctraining/scRNA-seq/blob/master/lessons/mitoRatio.md

# Connect to AnnotationHub
# the snapshot is saved at /home/aahn/.cache/R/AnnotationHub
ah &lt;- AnnotationHub()

# Access the Ensembl database for organism
ahDb &lt;- query(ah, 
              pattern = c(&quot;Homo sapiens&quot;, &quot;EnsDb&quot;), 
              ignore.case = TRUE)

#Next, we acquire the latest annotation files from this Ensembl database.
#We can first check which annotation versions are available:
# Check versions of databases available
ahDb %&gt;% 
  mcols()

# Since we want the most recent, we will return the AnnotationHub ID for this database:
# Acquire the latest annotation files
id &lt;- ahDb %&gt;%
  mcols() %&gt;%
  rownames() %&gt;%
  tail(n = 1)

#Finally, we can use the AnnotationHub connection to download the appropriate Ensembl database, which should be version GRCh38.92.

# Download the appropriate Ensembldb database
# this took ages so will need to save the rds 
edb &lt;- ah[[id]]

# this did not work ... i guess you cant save a ENsDb object? 
#saveRDS(edb, &quot;/researchers/antonio.ahn/resources/R_resources/annotationhub/data/edb_AH104864.rds&quot;)

#And to extract gene-level information we can use the Ensembldb function genes() to return a data frame of annotations.

# Extract gene-level information from database
annotations &lt;- genes(edb, 
                     return.type = &quot;data.frame&quot;)       

# saved as these steps takes quite a while
saveRDS(annotations, &quot;/researchers/antonio.ahn/resources/R_resources/annotationhub/data/annotations_AH104864.rds&quot;)
</code></pre>
<pre><code class="language-r">annoHub &lt;- readRDS(&quot;/researchers/antonio.ahn/resources/R_resources/annotationhub/data/annotations_AH104864.rds&quot;)
</code></pre>
<pre><code class="language-r">index &lt;- match(names(geneList), annoHub$entrezid)
annoHub_gs &lt;- annoHub[index,] %&gt;% .$symbol
</code></pre>
<h1 id="mapids">mapIds</h1>
<pre><code class="language-r">library(org.Hs.eg.db)
</code></pre>
<pre><code class="language-r">mapIds_gs &lt;- mapIds(org.Hs.eg.db, keys=names(geneList), keytype=&quot;ENTREZID&quot;, columns=&quot;SYMBOL&quot;,column=&quot;SYMBOL&quot;)
</code></pre>
<h1 id="summary">summary</h1>
<p>in this example, the mapID method gave the most non NA&rsquo;s</p>
<pre><code class="language-r">summary_df &lt;- tibble(entrezID = names(geneList), biomart_gs = biomart_gs, mapIds_gs, annoHub_gs)
</code></pre>
<pre><code class="language-r">summary_df$biomart_gs %&gt;% is.na %&gt;% summary
</code></pre>
<pre><code>##    Mode   FALSE    TRUE 
## logical   12263     232
</code></pre>
<pre><code class="language-r">summary_df$mapIds_gs %&gt;% is.na %&gt;% summary
</code></pre>
<pre><code>##    Mode   FALSE    TRUE 
## logical   12440      55
</code></pre>
<pre><code class="language-r">summary_df$annoHub_gs %&gt;% is.na %&gt;% summary
</code></pre>
<pre><code>##    Mode   FALSE    TRUE 
## logical   12214     281
</code></pre>

</main>

  <footer>
  <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/katex/dist/katex.min.css">
<script src="//cdn.jsdelivr.net/combine/npm/katex/dist/katex.min.js,npm/katex/dist/contrib/auto-render.min.js,npm/@xiee/utils/js/render-katex.js" defer></script>

<script src="//cdn.jsdelivr.net/npm/@xiee/utils/js/center-img.min.js" defer></script>

  
  <hr/>
  <a href="https://github.com/AntonioAhn">Github</a> | <a href="https://twitter.com/AntonioAhnNZ">Twitter</a>
  
  </footer>
  </body>
</html>

