<!DOCTYPE html>
<html lang="en-us">
  <head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=443&amp;path=livereload" data-no-instant defer></script>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>venn_diagram | Antonio Ahn&#39;s website</title>
    <link rel="stylesheet" href="../../css/style.css" />
    <link rel="stylesheet" href="../../css/fonts.css" />
    
  </head>

  <body>
    <nav>
    <ul class="menu">
      
      <li><a href="../../">Home</a></li>
      
      <li><a href="../../posts/">Posts</a></li>
      
      <li><a href="../../about/">About</a></li>
      
      <li><a href="../../categories/">Categories</a></li>
      
      <li><a href="../../tags/">Tags</a></li>
      
      <li><a href="../../publication/">Publications</a></li>
      
      <li><a href="../../resume/">Resume</a></li>
      
    </ul>
    <hr/>
    </nav>

<div class="article-meta">
<h1><span class="title">venn_diagram</span></h1>
<h2 class="author">antonio ahn</h2>
<h2 class="date">2023/10/14</h2>
</div>

<main>
<h1 id="load-library">load library</h1>
<pre><code class="language-r">library(rtracklayer)
library(tidyverse)
library(ChIPpeakAnno)
library(ggsci)
</code></pre>
<h1 id="load-peak-files">load peak files</h1>
<p>From &ldquo;The ChIPpeakAnno userâ€™s guide&rdquo;</p>
<pre><code class="language-r">bed &lt;- system.file(&quot;extdata&quot;, &quot;MACS_output.bed&quot;, package=&quot;ChIPpeakAnno&quot;)
gr1 &lt;- toGRanges(bed, format=&quot;BED&quot;, header=FALSE)

gff &lt;- system.file(&quot;extdata&quot;, &quot;GFF_peaks.gff&quot;, package=&quot;ChIPpeakAnno&quot;)
gr2 &lt;- toGRanges(gff, format=&quot;GFF&quot;, header=FALSE, skip=3)
</code></pre>
<pre><code>## If you are importing files downloaded from ensembl, 
##           it will be better to import the files into a TxDb object,
##           and then convert to GRanges by toGRanges. Here is the sample code:
##           library(GenomicFeatures)
##           txdb &lt;- makeTxDbFromGFF('/config/binaries/R/4.2.0/R_libraries/ChIPpeakAnno/extdata/GFF_peaks.gff')
##           anno &lt;- toGRanges(txdb, format='gene')
</code></pre>
<pre><code class="language-r">ol &lt;- findOverlapsOfPeaks(gr1, gr2)
</code></pre>
<pre><code>## duplicated or NA names found. 
##                 Rename all the names by numbers.
</code></pre>
<h1 id="make-venn-diagram">make venn diagram</h1>
<pre><code class="language-r">col_npg5 &lt;- c(ggsci::pal_npg()(10)[c(3,4,9,1,2)])

# don't write log file for VennDiagram
futile.logger::flog.threshold(futile.logger::ERROR, name = &quot;VennDiagramLogger&quot;)

makeVennDiagram(ol,
                fill=col_npg5[1:2], # circle fill color
                col=col_npg5[1:2], #circle border color
                #connectedPeaks = &quot;merge&quot;, connectedPeaks = &quot;keepAll&quot;, min keepFirstListConsistent
                cat.col=col_npg5[1:2], connectedPeaks = &quot;merge&quot;, scaled=T, by = &quot;region&quot;, main = &quot;venn&quot;, NameOfPeaks = c(&quot;control&quot;,&quot;treatment_1&quot;), cat.cex = 1.5, cat.pos = c(0, 0))  
</code></pre>
<img src="../../posts/2023-10-14-venn-diagram/index_files/figure-html/unnamed-chunk-3-1.png" width="672" />

</main>

  <footer>
  <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/katex/dist/katex.min.css">
<script src="//cdn.jsdelivr.net/combine/npm/katex/dist/katex.min.js,npm/katex/dist/contrib/auto-render.min.js,npm/@xiee/utils/js/render-katex.js" defer></script>

<script src="//cdn.jsdelivr.net/npm/@xiee/utils/js/center-img.min.js" defer></script>

  
  <hr/>
  <a href="https://github.com/AntonioAhn">Github</a> | <a href="https://twitter.com/AntonioAhnNZ">Twitter</a>
  
  </footer>
  </body>
</html>

